### PR TITLE
VEGA-258 Change password

### PR DESCRIPTION
**Pull request by hawx**

### CODE REVIEW

File: internal/server/change_password.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+			if err == sirius.ErrUnauthorized {
+				http.Redirect(w, r, siriusURL+"/auth", http.StatusFound)
+				return
+			} else if _, ok := err.(sirius.ClientError); ok {
+				http.Redirect(w, r, prefix+"/change-password?error="+url.QueryEscape(err.Error()), http.StatusFound)
+				return
+			} else if err != nil {
+				logger.Println("changePassword:", err)
+				http.Redirect(w, r, prefix+"/change-password", http.StatusFound)
```
Comment (gregtyler) [inline internal/server/change_password.go:55]: Should probably surface a generic error here, so the user knows their changes weren't saved

File: web/template/change-password.gotmpl [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    <div class="govuk-grid-column-two-thirds">
+      <h1 class="govuk-heading-xl">Change password</h1>
+
+      {{ if .Error }}
+        <div class="govuk-error-summary" aria-labelledby="error-summary-title" role="alert" tabindex="-1" data-module="govuk-error-summary">
+          <h2 class="govuk-error-summary__title" id="error-summary-title">There was a problem</h2>
+
+          <div class="govuk-error-summary__body">
+            <ul class="govuk-list govuk-error-summary__list">
+              <li>{{ .Error }}</li>
```
Comment (gregtyler) [inline web/template/change-password.gotmpl:16]: Per [Design System](https://design-system.service.gov.uk/components/error-summary/#linking-from-the-error-summary-to-each-answer) "If you do not know which field contains an error, link to the first field."
```suggestion
              <li>
                <a href="#f-currentpassword">{{ .Error }}</a>
              </li>
```

Comment (hawx) [reply web/template/change-password.gotmpl:16]: I have obviously never really read this page, this is weird advise but OK, also spotted

> As well as showing an error summary, follow the validation pattern - for example, by adding ‘Error: ’ to the beginning of the page <title> so screen readers read it out as soon as possible.

File: internal/server/change_password.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+type changePasswordVars struct {
+	Path      string
+	SiriusURL string
+	Prefix    string
+	Error     string
+}
+
+func changePassword(logger *log.Logger, client ChangePasswordClient, tmpl Template, prefix, siriusURL string) http.Handler {
```
[the remark this answers is not in the document]
Comment (hawx) [reply internal/server/change_password.go:21]: I'd always try to pass the data down from the top-level, having random places getting info from the environment will always get messy. I've added a template function to add the prefix to a string instead, so we don't need to pass it to every template. That still does leave it in some handlers as they have to do a redirect to a route, maybe we should use relative URLs?

[the remark this answers is not in the document]
Comment (gregtyler) [reply internal/server/change_password.go:21]: Cool, and some of that will be further removed by the `RedirectError` change 👍

**Review by gregtyler — APPROVED** [review]

File: web/template/change-password.gotmpl (hunk 2 above)
Comment (gregtyler) [reply web/template/change-password.gotmpl]: Good catch, I've read that a bunch of times and never done it.

**Pull request merged**